### ISSUE TITLE
`TestOptionalInputArtifacts ` on `master` build broken

### ISSUE THREAD

**Issue by alexec** [argoproj/argo-workflows#8412]
https://github.com/argoproj/argo-workflows/actions/workflows/ci-build.yaml?query=branch%3Amaster

![image](https://user-images.githubusercontent.com/1142830/164032567-e6e20ff0-17a5-41ab-981f-9fc3ea3e0cea.png)

```
=== RUN   TestFunctionalSuite/TestOptionalInputArtifacts
[1107](https://github.com/argoproj/argo-workflows/runs/6067635095?check_suite_focus=true#step:16:1107)
Submitting workflow  input-artifacts-
[1108](https://github.com/argoproj/argo-workflows/runs/6067635095?check_suite_focus=true#step:16:1108)
Waiting 1m0s for workflow metadata.name=input-artifacts-dqncz
[1109](https://github.com/argoproj/argo-workflows/runs/6067635095?check_suite_focus=true#step:16:1109)
 ? input-artifacts-dqncz Workflow 0s      
[1110](https://github.com/argoproj/argo-workflows/runs/6067635095?check_suite_focus=true#step:16:1110)

[1111](https://github.com/argoproj/argo-workflows/runs/6067635095?check_suite_focus=true#step:16:1111)
 ● input-artifacts-dqncz   Workflow 0s      
[1112](https://github.com/argoproj/argo-workflows/runs/6067635095?check_suite_focus=true#step:16:1112)
 └ ◷ input-artifacts-dqncz Pod      0s      
[1113](https://github.com/argoproj/argo-workflows/runs/6067635095?check_suite_focus=true#step:16:1113)

[1114](https://github.com/argoproj/argo-workflows/runs/6067635095?check_suite_focus=true#step:16:1114)
 ● input-artifacts-dqncz   Workflow 0s      
[1115](https://github.com/argoproj/argo-workflows/runs/6067635095?check_suite_focus=true#step:16:1115)
 └ ◷ input-artifacts-dqncz Pod      0s      PodInitializing
[1116](https://github.com/argoproj/argo-workflows/runs/6067635095?check_suite_focus=true#step:16:1116)

[1117](https://github.com/argoproj/argo-workflows/runs/6067635095?check_suite_focus=true#step:16:1117)
 ⚠ input-artifacts-dqncz   Workflow 2s      Error (exit code 1): runtime error: invalid memory address or nil pointer dereference
[1118](https://github.com/argoproj/argo-workflows/runs/6067635095?check_suite_focus=true#step:16:1118)
 └ ⚠ input-artifacts-dqncz Pod      1s      Error (exit code 1): runtime error: invalid memory address or nil pointer dereference
[1119](https://github.com/argoproj/argo-workflows/runs/6067635095?check_suite_focus=true#step:16:1119)

[1120](https://github.com/argoproj/argo-workflows/runs/6067635095?check_suite_focus=true#step:16:1120)
Condition "to be done" met after 2s
[1121](https://github.com/argoproj/argo-workflows/runs/6067635095?check_suite_focus=true#step:16:1121)
Checking expectation input-artifacts-dqncz
[1122](https://github.com/argoproj/argo-workflows/runs/6067635095?check_suite_focus=true#step:16:1122)
input-artifacts-dqncz : Error Error (exit code 1): runtime error: invalid memory address or nil pointer dereference
[1123](https://github.com/argoproj/argo-workflows/runs/6067635095?check_suite_focus=true#step:16:1123)
    functional_test.go:574: 
[1124](https://github.com/argoproj/argo-workflows/runs/6067635095?check_suite_focus=true#step:16:1124)
        	Error Trace:	functional_test.go:574
[1125](https://github.com/argoproj/argo-workflows/runs/6067635095?check_suite_focus=true#step:16:1125)
        	            				then.go:67
[1126](https://github.com/argoproj/argo-workflows/runs/6067635095?check_suite_focus=true#step:16:1126)
        	            				then.go:42
[1127](https://github.com/argoproj/argo-workflows/runs/6067635095?check_suite_focus=true#step:16:1127)
        	            				functional_test.go:573
[1128](https://github.com/argoproj/argo-workflows/runs/6067635095?check_suite_focus=true#step:16:1128)
        	Error:      	Not equal: 
[1129](https://github.com/argoproj/argo-workflows/runs/6067635095?check_suite_focus=true#step:16:1129)
        	            	expected: "Succeeded"
[1130](https://github.com/argoproj/argo-workflows/runs/6067635095?check_suite_focus=true#step:16:1130)
        	            	actual  : "Error"
[1131](https://github.com/argoproj/argo-workflows/runs/6067635095?check_suite_focus=true#step:16:1131)
        	            	
[1132](https://github.com/argoproj/argo-workflows/runs/6067635095?check_suite_focus=true#step:16:1132)
        	            	Diff:
[1133](https://github.com/argoproj/argo-workflows/runs/6067635095?check_suite_focus=true#step:16:1133)
        	            	--- Expected
[1134](https://github.com/argoproj/argo-workflows/runs/6067635095?check_suite_focus=true#step:16:1134)
        	            	+++ Actual
[1135](https://github.com/argoproj/argo-workflows/runs/6067635095?check_suite_focus=true#step:16:1135)
        	            	@@ -1,2 +1,2 @@
[1136](https://github.com/argoproj/argo-workflows/runs/6067635095?check_suite_focus=true#step:16:1136)
        	            	-(v1alpha1.WorkflowPhase) (len=9) "Succeeded"
[1137](https://github.com/argoproj/argo-workflows/runs/6067635095?check_suite_focus=true#step:16:1137)
        	            	+(v1alpha1.WorkflowPhase) (len=5) "Error"
[1138](https://github.com/argoproj/argo-workflows/runs/6067635095?check_suite_focus=true#step:16:1138)
        	            	 
[1139](https://github.com/argoproj/argo-workflows/runs/6067635095?check_suite_focus=true#step:16:1139)
        	Test:       	TestFunctionalSuite/TestOptionalInputArtifacts
[1140](https://github.com/argoproj/argo-workflows/runs/6067635095?check_suite_focus=true#step:16:1140)
=== FAIL: FunctionalSuite/TestOptionalInputArtifacts
```

**Comment by alexec** [comment]
Caused by feat: added support for http as an option for artifact upload. Fixes #785 (#8405)

**Comment by alexec** [comment]
Reverted 5845efbb9..e71fdee07
